universal_transfer_operator/src/universal_transfer_operator/data_providers/database/sqlite.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+import socket
+
+import attr
+from airflow.providers.sqlite.hooks.sqlite import SqliteHook
+from sqlalchemy import MetaData as SqlaMetaData, create_engine
+from sqlalchemy.engine.base import Engine
+from sqlalchemy.sql.schema import Table as SqlaTable
+
+from universal_transfer_operator.data_providers.database.base import DatabaseDataProvider, FileStream
+from universal_transfer_operator.datasets.table import Metadata, Table
+from universal_transfer_operator.universal_transfer_operator import TransferParameters
+
+
+class SqliteDataProvider(DatabaseDataProvider):
+    """SqliteDataProvider represent all the DataProviders interactions with Sqlite Databases."""
+
+    def __init__(
+        self,
+        dataset: Table,
+        transfer_mode,
+        transfer_params: TransferParameters = attr.field(
+            factory=TransferParameters,
+            converter=lambda val: TransferParameters(**val) if isinstance(val, dict) else val,
+        ),
+    ):
+        self.dataset = dataset
+        self.transfer_params = transfer_params
+        self.transfer_mode = transfer_mode
+        self.transfer_mapping = {}
+        self.LOAD_DATA_NATIVELY_FROM_SOURCE: dict = {}
+        super().__init__(
+            dataset=self.dataset, transfer_mode=self.transfer_mode, transfer_params=self.transfer_params
+        )
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(conn_id="{self.dataset.conn_id})'
+
+    @property
+    def sql_type(self) -> str:
+        return "sqlite"
+
+    @property
+    def hook(self) -> SqliteHook:
+        """Retrieve Airflow hook to interface with the Sqlite database."""
+        return SqliteHook(sqlite_conn_id=self.dataset.conn_id)
+
+    @property
+    def sqlalchemy_engine(self) -> Engine:
+        """Return SQAlchemy engine."""
+        # Airflow uses sqlite3 library and not SqlAlchemy for SqliteHook
+        # and it only uses the hostname directly.
+        airflow_conn = self.hook.get_connection(self.dataset.conn_id)
+        return create_engine(f"sqlite:///{airflow_conn.host}")
+
+    @property
+    def default_metadata(self) -> Metadata:
+        """Since Sqlite does not use Metadata, we return an empty Metadata instances."""
+        return Metadata()
+
+    def read(self):
+        """ ""Read the dataset and write to local reference location"""
+        raise NotImplementedError
+
+    def write(self, source_ref: FileStream):
+        """Write the data from local reference location to the dataset"""
+        return self.load_file_to_table(
+            input_file=source_ref.actual_file,
+            output_table=self.dataset,
+        )
```

Comment:
@ utkarsharma2 I would like to have simple tests for all these functions - are these the ones you said are not needed in the context of UTO? If they are not necessary, perhaps we should just remove them for now.

Reply:
@tatiana I meant the signature of the methods might change since we already keep a reference to a dataset in the Dataset provider object.